src/repl/handler.rs

```diff
@@ -154,22 +157,19 @@ impl ReplyStreamHandler {
     }
 
     pub fn done(&mut self) -> Result<()> {
-        match self.sender.as_ref() {
-            Some(tx) => {
-                let ret = tx
-                    .send(ReplyStreamEvent::Done)
-                    .with_context(|| "Failed to send StreamEvent:Done");
-                self.safe_ret(ret)?;
+        if let Some(tx) = self.sender.as_ref() {
+            let ret = tx
+                .send(ReplyStreamEvent::Done)
+                .with_context(|| "Failed to send StreamEvent:Done");
+            self.safe_ret(ret)?;
+        } else {
+            if !self.buffer.ends_with('\n') {
+                print_now!("\n");
             }
-            None => {
-                if !self.buffer.ends_with('\n') {
-                    print_now!("\n")
-                }
-                if self.repl {
+            if self.repl {
+                print_now!("\n");
+                if cfg!(macos) {
                     print_now!("\n");
-                    if cfg!(macos) {
-                        print_now!("\n")
-                    }
```

Comment:
almost purely stylistic, but if let causes less nesting and is more readable in this case.